src/parser/expression.js

```diff
@@ -827,11 +827,7 @@ pp.parseObjPropValue = function (prop, startPos, startLoc, isGenerator, isAsync,
 
   if (!prop.computed && prop.key.type === "Identifier") {
     if (isPattern) {
-      if (this.isKeyword(prop.key.name)) {
```

Comment:
Moved this into the `checkReservedWord()` method, since it should be with the other checks.